apps/backend/src/controllers/neighborhoods.ts

```diff
@@ -105,4 +107,102 @@ neighborhoodsRouter.post('/:id/join', middleware.userIdExtractorAndLoginValidato
+  const neighborhoodID = Number(req.params.id);
+  // LoginValidator ensures that loggedUserId is present
+  const loggedUserID = req.loggedUserId as number;
+
+  const isUserMemberOfNeighborhood: boolean = await neighborhoodServices
+    .isUserMemberOfNeighborhood(loggedUserID, neighborhoodID);
+
+  if (!isUserMemberOfNeighborhood) {
+    return res.status(400).send({ error: 'user is not a member of the neighborhood' });
+  }
+
+  const requests: RequestData[] = await neighborhoodServices
+    .getRequestsAssociatedWithNeighborhood(neighborhoodID);
+
+  return res.status(200).send(requests);
+}));
+
+// Get a single request
+neighborhoodsRouter.get('/:neighborhoodId/requests/:requestId', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const requestId = Number(req.params.requestId);
+  const neighborhoodId = Number(req.params.neighborhoodId);
+  const loggedUserId = req.loggedUserId as number;
+
+  const isUserMemberOfNeighborhood = await neighborhoodServices
+    .isUserMemberOfNeighborhood(loggedUserId, neighborhoodId);
+
+  if (!isUserMemberOfNeighborhood) {
+    return res.status(401).send({ error: 'user not a member of neighborhood' });
+  }
+
+  const isRequestAssociatedWithNeighborhood = await neighborhoodServices
+    .isRequestAssociatedWithNeighborhood(requestId, neighborhoodId);
+
+  if (!isRequestAssociatedWithNeighborhood) {
+    return res.status(400).send({ error: 'request not associated with the neighborhood' });
+  }
+
+  const request: RequestData = await requestServices.getRequestById(requestId);
+  return res.status(200).send(request);
+}));
+
+// Create request
+neighborhoodsRouter.post('/:neighborhoodId/requests', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const postData: CreateRequestData = await requestServices.parseCreateRequestData(req.body);
+
+  const loggedUserId: number = req.loggedUserId as number;
+  const neighborhoodId: number = Number(req.params.neighborhoodId) as number;
+
+  const request: RequestData = await neighborhoodServices
+    .createRequest(postData, loggedUserId, neighborhoodId);
+
+  return res.status(201).send(request);
+}));
+
+// Update request
+neighborhoodsRouter.put(
+  '/:neighborhoodId/requests/:requestId/',
+  middleware.userIdExtractorAndLoginValidator,
+  catchError(async (req: RequestWithAuthentication, res: Response) => {
+    const userId = Number(req.loggedUserId);
+    const { requestId, neighborhoodId } = req.params;
+
+    const isOwnerUser = await requestServices.hasUserCreatedRequest(+requestId, userId);
+    if (!isOwnerUser) {
+      return res.sendStatus(401);
+    }
+
+    const updatedRequest: RequestData = await requestServices.updateRequest(
+      req.body,
+      +requestId,
+      +neighborhoodId,
+    );
+
+    return res.status(200).json(updatedRequest);
+  }),
+);
+
+// Delete request
+neighborhoodsRouter.delete(
```

Comment:
Same with delete. This is getting a bit confusing.

Now requests are data in their own right as well along with being sub-data of the neighborhood. So accessing them both as `/neighborhood/:id/requests/:id` and `/requests/:id` are correct. 

I am not sure about it. :-/

Reply:
I'm not a fan of having both, either. I initially thought `/neighborhood/:id/requests/:id` would be better to obtain the neighborhood id, but if we find that we don't need it as request have unique ids across all the db, we can revert to the `/api/requests` api.

Reply:
My feeling is that the only time that we need this structure with the two `:id` values is when we are working with the many-to-many relation, or users/neighborhoods. As for where to put the request delete and update, my first instinct is to have them in request.ts. The neighborhood id is part of the request table, gets added on creation and will always be there. I think it's important to make that field read-only, though. We don't need to reassign requests from one neighborhood to another.

The create request route should capture the current neighborhood id and plug it into the JSON for the request data. Exactly how to do that isn't immediately obvious to me, but perhaps you guys already know exactly how to do that.

Reply:
Done.